components/dashboard/src/settings/Teams.tsx

```diff
@@ -0,0 +1,856 @@
+    const [isStudent, setIsStudent] = useState<boolean>(false);
+    const [teamSubscriptions, setTeamSubscriptions] = useState<TeamSubscription[]>([]);
+
+    const [createTeamModal, setCreateTeamModal] = useState<{ types: string[], defaultCurrency: string } | undefined>(undefined);
+    const [manageTeamModal, setManageTeamModal] = useState<{ sub: TeamSubscription } | undefined>(undefined);
+    const [inviteMembersModal, setInviteMembersModal] = useState<{ sub: TeamSubscription } | undefined>(undefined);
+    const [addMembersModal, setAddMembersModal] = useState<{ sub: TeamSubscription } | undefined>(undefined);
+
+    const [pendingPlanPurchase, setPendingPlanPurchase] = useState<{ plan: Plan } | undefined>(getLocalStorageObject<{ plan: Plan }>("pendingPlanPurchase"));
+    const [pendingSlotsPurchase, setPendingSlotsPurchase] = useState<{ ts: TeamSubscription } | undefined>(getLocalStorageObject<{ ts: TeamSubscription }>("pendingSlotsPurchase"));
+
+    useEffect(() => {
+        queryState();
+    }, []);
+
+    useEffect(() => {
+        if (pendingPlanPurchase) {
+            setLocalStorageObject("pendingPlanPurchase", pendingPlanPurchase);
+        } else {
+            removeLocalStorageObject("pendingPlanPurchase");
+        }
+    }, [pendingPlanPurchase]);
+
+    useEffect(() => {
+        if (pendingSlotsPurchase) {
+            setLocalStorageObject("pendingSlotsPurchase", pendingSlotsPurchase);
+        } else {
+            removeLocalStorageObject("pendingSlotsPurchase");
+        }
+    }, [pendingSlotsPurchase]);
+
+    const queryState = async () => {
+        try {
+            const [slots, showPaymentUI, teamSubscriptions, clientRegion, isStudent] = await Promise.all([
+                getGitpodService().server.tsGetSlots(),
+                getGitpodService().server.getShowPaymentUI(),
+                getGitpodService().server.tsGet(),
+                getGitpodService().server.getClientRegion(),
+                getGitpodService().server.isStudent(),
+            ]);
+    
+            setDefaultCurrency((clientRegion && (countries as any)[clientRegion]?.currency === 'EUR') ? 'EUR' : 'USD');
+            setIsStudent(isStudent);
+    
+            setSlots(slots);
+            setShowPaymentUI(showPaymentUI);
+            setTeamSubscriptions(teamSubscriptions);
+        } catch (error) {
+            console.log(error);
+        }
+    }
+
+    useEffect(() => {
+        if (!isChargebeeCustomer) {
+            (async () => {
+                try {
+                    setIsChargebeeCustomer(await getGitpodService().server.isChargebeeCustomer());
+                } catch (error) {
+                    console.log(error);
+                }
+            })()
+        }
+        if (pendingPlanPurchase) {
+            if (teamSubscriptions.some(ts => ts.planId === pendingPlanPurchase.plan.chargebeeId)) {
+                setPendingPlanPurchase(undefined);
+            }
```

Comment:
Also, if you've just loaded the page, and there is a `pendingPlanPurchase`, I think any previous polling is no longer active.

In that case, I think you'd need to resume polling here:

```suggestion
            }
            pollForPlanPurchased(pendingPlanPurchase.plan);
```